src/field/pokemon.ts

```diff
@@ -1259,6 +1259,11 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
       }
     }
 
+    // the type added to Pokemon from moves like Forest's Curse or Trick Or Treat
+    if (this.summonData && this.summonData.addedType && !types.includes(this.summonData.addedType)) {
```

Comment:
```suggestion
    if (!ignoreOverride && this.summonData && this.summonData.addedType && !types.includes(this.summonData.addedType)) {
```
`ignoreOverride` should probably also ignore overriding types with Forest's Curse and Trick Or Treat